docs/reconciler-proposal/kris-1/interface.go

```diff
@@ -0,0 +1,62 @@
+// Copyright © 2017 The Kubicorn Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloud
+
+import (
+	"github.com/kris-nova/kubicorn/apis/cluster"
+)
+
+// Model will model resources to an API representation of infrastructure
+type Model interface {
+
+	// Get will return the map of ordered resources against a cloud
+	Get() map[int]Resource
```

Comment:
Order is critical here - and building a hash map is one way of preserving order... we go `0:N` for a create and `N:0` for a delete. 

By using a map instead of a slice, we can index them and use the indexes in an arbitrary way to point to resources down the line as seen here https://github.com/kris-nova/kubicorn/blob/master/cloud/amazon/model.go#L43